src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,85 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_SIZE = 16;
+    private static final double RESIZE_BOUND = 0.75;
+    private int size;
+    private int threshold;
+    private Node<K, V>[] nodeArray;
+
+    public MyHashMap() {
+        nodeArray = new Node[DEFAULT_SIZE];
+        threshold = (int) (nodeArray.length * RESIZE_BOUND);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= threshold) {
+            resize();
+        }
+        Node<K, V> newNode = new Node<>((K) key, (V) value, null);
```

Comment:
There's no need to cast 'key' and 'value' to 'K' and 'V' types respectively as they are already of that type. The correct instantiation should be 'new Node<>(key, value, null);'.